src/io/network/endpoint.cpp

```diff
@@ -40,26 +41,24 @@ Endpoint::IpFamily Endpoint::GetIpFamily(const std::string &ip_address) {
 std::optional<std::pair<std::string, uint16_t>> Endpoint::ParseSocketOrIpAddress(
     const std::string &address, const std::optional<uint16_t> default_port = {}) {
   /// expected address format:
-  ///   - "ip_address:port_number"
-  ///   - "ip_address"
+  ///   - "address:port_number"
```

Comment:
Do not change the behaviour of `ParseSocketOrIpAddress` 
Introduce a new method for different behaviour.